model/task/task.go

```diff
@@ -1580,6 +1580,13 @@ func (t *Task) BlockedState() (string, error) {
 		return "", err
 	}
 
+	return t.blockedStatePrivate()
+}
+
+func (t *Task) blockedStatePrivate() (string, error) {
+	if len(t.DependsOn) == 0 {
```

Comment:
Is https://github.com/evergreen-ci/evergreen/pull/1613/commits/c64ae94f6e04909288ace61d7df2df0267b94e7b#diff-ec70289495c0b3b32aa040cc657874cfR1576 now redundant?

Reply:
Not necessarily - it will save us from having to do the query for tasks with no dependencies